### ISSUE TITLE
Automatic triggering of custom analyzer

### ISSUE THREAD

**Issue by YakupIpek21** [mrlacey/Rapid-XAML-Toolkit#505]
**Question**
Is there a way to automatically trigger the custom analyzer when a project containing a xaml is built?
Not only when the file is saved.

**Desired Behaviour**
So any project referencing a custom analyzer should automatically analyze its existing Xaml files.

**Comment by mrlacey** [comment]
Yes, add [this NuGet package](https://www.nuget.org/packages/RapidXaml.BuildAnalysis/) to the project.

It adds a build step that runs the analysis.
The intention is to allow you to have XAML analysis of a project even if the extension is not installed. (Such as on a build server 😉)


Note that the currently available version also suffers from the issue identified in #503 and so a new version of the `RapidXaml.BuildAnalysis` package is also needed.

**Comment by YakupIpek21** [comment]
I see, thank you very much for the explanation

**Comment by mrlacey** [comment]
Version 0.13.0 of this NuGet package is now available and fixes the issue noted in #503 
Example:
![image](https://user-images.githubusercontent.com/189547/153283292-028bb943-5378-447b-847b-cfbc441af770.png)


**PLEASE NOTE. This currently only works with projects that use the old internal structure. It does not yet work with SDK-style projects (As you'll get if you create a project targeting .NET5 or 6.**
The changes to support SDK style projects is being tracked separately in #408

**Comment by mrlacey** [comment]
If you have any problems, comments, or suggestions please raise another issue.
If you love it, please [leave a rating/review](https://marketplace.visualstudio.com/items?itemName=MattLaceyLtd.RapidXamlToolkit&ssr=false#review-details) or consider [sponsorship](https://github.com/sponsors/mrlacey)  :smile:

**Comment by YakupIpek21** [comment]
After having added the RapidXaml.BuildAnalysis nuget package to the project which contains the xaml files, it does not produce any warnings when the project is built in VisualStudio 19

**Expected Behaviour**
When building a project which contains xaml files, the analysis should be triggered and warnings should be displayed


**To reproduce**
1. Generate a custom analyzer
![image](https://user-images.githubusercontent.com/7056714/153581521-77ed228d-6d5c-4a85-95e8-0908cbc8c5a8.png)

2. Add the project as a reference to the project containing xaml files (let's denote it as project **A**)
3. Add the RapidXaml.BuildAnalysis nuget package (Version 0.13.0) to the project **A**
4. In project **A** add a **TextBlock** Tag without the **IsEnabled** Attribute, save the file
5. Enable in the options the **Enable Custom Analysis** menu
6. When a warning is shown then restart VS19, so that no warning is shown, now build project **A**
7. In my case no warning is shown in VS19

**Comment by mrlacey** [comment]
> After having added the RapidXaml.BuildAnalysis nuget package to the project which contains the xaml files, it does not produce any warnings when the project is built in VisualStudio 19

What type of project are you using?
Can you share a complete reproduction project, not just a partial description.


Also, please note, from [above](https://github.com/mrlacey/Rapid-XAML-Toolkit/issues/505#issuecomment-1034160119):

> PLEASE NOTE. This currently only works with projects that use the old internal structure. It does not yet work with SDK-style projects (As you'll get if you create a project targeting .NET5 or 6.)
The changes to support SDK style projects is being tracked separately in https://github.com/mrlacey/Rapid-XAML-Toolkit/issues/408

**Comment by YakupIpek21** [comment]
Yes sure. The following reproduction steps refer to Visual Studio 2019 Version 16.11.5

1. Via the **Custom Rapid XAML Analyzer** project template I am creating a custom analyzer with the following content
![image](https://user-images.githubusercontent.com/7056714/153617529-34fd4a9d-9ece-40e5-8f05-226417b4b483.png)
2. In the same solution I am creating a WPF-App with the target framework **.Net Core 3.1**
3. The analyzer from step 1 is added as a reference to the WPF-app
4. The RapidXaml.BuildAnalysis nuget package (Version 0.13.0) is added via the nuget manager to the WPF-app 
5. The menu entry **Enable Custom Analysis** is enabled
6. In the WPF-app MainWindow.xaml file I added a TextBlock tag within the Grid tag and saved the xaml file
7. I restart VS19, since in step 6 the analyzer is triggered through the saving
8. I close all tabs in VS19 and build the WPF-app
9. When I check for the warnings nothing is displayed
10. When I open the MainWindow.xaml the warning is shown (not always)


**Expected Behaviour**
When building the project in Visual Studio 2019 the warnings should be displayed in the **Error List Window** without needing to open or saving the xaml file in Visual Studio 2019.

**Comment by mrlacey** [comment]
Unfortunately, WPF with .Net Core 3.1 also uses an SDK style project format and so the BuildAnalysis doesn't know which files to check. This is why it is not reporting anything.

I'll look to prioritize supporting this.  (Very few people have been using RXT with WPF projects and so this is a good opportunity to improve this experience.)

**Comment by mrlacey** [comment]
X-Ref #518
